modules/core/src/smithy4s/http/CodecAPI.scala

```diff
@@ -40,7 +40,7 @@ trait CodecAPI {
     * @param schema the value's schema
     * @return the codec associated to the A value.
     */
-  def compileCodec[A](schema: Schema[A]): Codec[A]
+  def compileCodec[A](schema: Schema[A], hintMask: HintMask): Codec[A]
```

Comment:
I don't really like this, the application of the hintMask should be an implementation detail of the `CodecAPI` instances I think